src/applications/gqrx/receiver.cpp

```diff
@@ -1110,16 +1114,20 @@ receiver::status receiver::stop_audio_playback()
 }
 
 /** Start UDP streaming of audio. */
-receiver::status receiver::start_udp_streaming(const std::string host, int port)
+receiver::status receiver::start_udp_streaming(const std::string host, int port, bool stereo)
 {
-    audio_udp_sink->start_streaming(host, port);
+    stop();
+    audio_udp_sink->start_streaming(host, port, stereo);
+    start();
```

Comment:
The addition of `stop()` and `start()` here has introduced a bug: pressing the UDP button will start up the receiver even if it was not running at the time.

Reply:
I don't think there's any need to stop and start the flow graph, you just need to add `lock()` and `unlock()` around the parts that reconfigure the connections in the UDP sink.